src/config.ts

```diff
@@ -65,6 +65,12 @@ export class KubeConfig implements SecurityAuthentication {
         new OpenIDConnectAuth(),
     ];
 
+    // Optionally add additional external authenticators, you must do this
+    // before you load a kubeconfig file that references them.
+    public static addAuthenticator(authenticator: Authenticator): void {
+        this.authenticators.push(authenticator);
```

Comment:
is the intended use here to add them to *all* existing KubeConfig objects since the `authenticators` property is static?

also, could we avoid the loading order issue by adding an optional parameter for an array of `authenticators` to the constructor instead? that way the authenticators must be added prior to loading a config

Reply:
I'm ok with this, but I also think it makes it a little more complicated. I wanted the authenticators to be able to self-register just by being `import`-ed which requires the static registration.  @cjihrig wdyt?